apache/superset
# build(helm): Helm-docs + fix Helm chart release

### SUMMARY

This PR implements 2 improvements to the releasing of Helm chart:
- Support for running [helm-docs](https://github.com/norwoodj/helm-docs) to generate a README for the chart, automatically generated from the `Chart.yaml` and annotated `values.yaml`. It will be enforced both as a `pre-commit` check, as well as a CI check.
- Fix the helm `chart-releaser` action which seems broken due to missing dependency on the bitnami chart repo (which prevented the release of the updated chart from #21806 ) - this may need some trial & error to confirm a successful release.

This PR includes a (useless) bump of the chart version to force the CI to go through a Helm release.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See [generated Chart](https://github.com/one-acre-fund/superset/tree/helm-docs/helm/superset) `README`:

![image](https://user-images.githubusercontent.com/59641363/198537267-8c4c6a1b-92f8-4ac3-b4c8-d8b839b65eb6.png)

### TESTING INSTRUCTIONS

- For the `helm-docs` pre-commit:
  - Check out project
  - Update pre-commit (`pre-commit install`)
  - Make some chart update (e.g. bump up the version in `Chart.yaml`) and stage the change
  - Try to commit. Git should raise an error complaining that a pre-commit check has failed
  - Run `pre-commit run helm-docs` to force execute the helm-docs generation, stage the resulting change and commit again
- For the chart releaser fix: merge the PR and hope for helm chart v0.7.6 to be released!

### ADDITIONAL INFORMATION
- [x] Has associated issue: #21806 (merge failed to release)
